File: demo/worker/worker/llm_prompt.py
"""Limit text sent to Hugging Face to control token cost."""

HF_MAX_TRANSCRIPT_CHARS = 14_000
HF_MAX_PROMPT_CHARS = 16_000


def truncate_for_llm(text: str, max_chars: int = HF_MAX_TRANSCRIPT_CHARS) -> str:
    t = text.strip()
    if len(t) <= max_chars:
        return t
    return f"{t[:max_chars]}\n\n[… truncated for model context limit …]"

File: demo/api/src/utils/llmPrompt.ts
/** Limit text sent to Hugging Face to control token cost. */
export const HF_MAX_TRANSCRIPT_CHARS = 14_000;
export const HF_MAX_ASSISTANT_USER_CHARS = 2_000;
export const HF_MAX_ASSISTANT_CONTEXT_CHARS = 12_000;

export function truncateForLlm(text: string, maxChars: number): string {
  const t = text.trim();
  if (t.length <= maxChars) return t;
  return `${t.slice(0, maxChars)}\n\n[… truncated for model context limit …]`;
}
